### PR TITLE
init: one-piece v0.1 - graphrag

### PR DESCRIPTION
**Pull request by bw-99**
## 개요

- one-piece v0.1 일부 구현 (graphrag, backend, frontend, devops 중 graphrag)

## 작업 내용

- openai 키 발급
- 경제적인 모델 탐색 (gpt-4o-mini가 제일 저렴함)
- 시범 운행
- [문서 작성](https://www.notion.so/TechSpec-one-piece-v0-1-16ad2f288baf80258765e52fa495065c?pvs=4)

### CODE REVIEW

**Comment by bw-99** [comment]
다음은 global query로 아래 사항을 물어본 결과.

<query>
What are the unique and representative core keywords associated with each research community in recommender systems (e.g., information retrieval, machine learning, data mining, HCI)? Please provide detailed keywords that are both well-representative and unique to each community. Additionally, could you suggest representative research papers for each community that highlight these keywords?

<answer>
# Core Keywords and Representative Research Papers in Recommender Systems

In the field of recommender systems, various research communities have developed unique terminologies and methodologies. Below is a detailed overview of the core keywords associated with each community, along with representative research papers that highlight these keywords.

## Information Retrieval Metrics Community
**Core Keywords:** Precision, Recall, NDCG, E-Measure, F-Measure  
These keywords are foundational for evaluating the effectiveness of search and recommendation systems.  
**Representative Research Papers:**  
- [Data: Reports (125, 254, 113)]

## Recommendation Systems Community
**Core Keywords:** Personalization, Collaborative Filtering, User Experience  
This community focuses on algorithms designed to suggest items based on user preferences.  
**Representative Research Papers:**  
- 'KGAT: Knowledge Graph Attention Network for Recommendation' [Data: Reports (218, 220, 115)]

## Printf Recommender System Community
**Core Keywords:** User Preference Modeling, Graph Learning Techniques, Item Images, Multimodal Approach  
These keywords emphasize advanced methodologies for enhancing recommendation accuracy through the integration of visual and textual data.  
**Representative Research Papers:**  
- 'Preference Modeling Based on User Reviews with Item Images and Textual Information via Graph Learning' [Data: Reports (80)]

## User-Item Engagement and Candidate Retrieval Systems Community
**Core Keywords:** User-Item Engagement Graph, Candidate Retrieval System, Bipartite Graph, Recall@K
This community focuses on user interactions and methodologies to enhance recommendation accuracy.
**Representative Research Papers:**
- [Data: Reports (91, 124, 176, 240, 205)]

## Collaborative Filtering Community
**Core Keywords:** User K-NN, Item K-NN, Slope One, FunkSVD
These keywords represent various algorithms used for personalized recommendations.
**Representative Research Papers:**
- 'Collaborative Filtering Methods' by Desrosiers and Karypis [Data: Reports (110)]

## Graph Convolutional Networks Community
**Core Keywords:** Graph Convolutional Neural Networks, Web-Scale Recommendation, Neural Network Architectures
This community emphasizes leveraging graph structures to improve recommendation processes.
**Representative Research Papers:**
- 'Graph Convolutional Neural Networks for Web-Scale Recommender Systems' [Data: Reports (105)]

## BERT Research Community
**Core Keywords:** BERT, Deep Learning, Text-Aware Preference Ranking
These keywords highlight the role of natural language processing in enhancing recommendation systems.
**Representative Research Papers:**
- 'BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding' [Data: Reports (215)]

## Advanced Models in Recommendation Systems Community
**Core Keywords:** NARRE, DeepCoNN, User Reviews, Attention Mechanisms
This community focuses on advanced models that enhance recommendation accuracy through user feedback.
**Representative Research Papers:**
- [Data: Reports (77)]

## Evaluation Methods Community
**Core Keywords:** Evaluation Metrics, SVD, Collaborative Filtering Techniques
This community emphasizes the assessment of recommendation algorithms.
**Representative Research Papers:**
- 'A Review of Recommender Systems Evaluation Methods' [Data: Reports (46)]

## User Engagement and Trust in Recommender Systems Community
**Core Keywords:** User Engagement, Trust Propagation, Relevance Variable
These keywords reflect the importance of user interactions and trust dynamics in tailoring recommendations.
**Representative Research Papers:**
- [Data: Reports (175, 66)]

## Conclusion
Each research community within the realm of recommender systems has developed its own unique set of keywords that reflect its focus and methodologies. The representative research papers cited provide a deeper understanding of how these keywords are applied in practice, showcasing the diversity and complexity of the field.

**Comment by bw-99** [comment]
0. 프로젝트 폴더에 example 폴더 생성
1. `python3 -m graphrag init --root ./example`
2. `python3 -m graphrag index --root ./example`
3. `python3 -m graphrag query --root ./example --method global --query "Please identify various domains in machine learning. What are the unique and representative core keywords associated with each research domain? Please provide detailed keywords that are both well-representative and unique to each domain. Additionally, could you suggest the title of representative research papers for each domain that highlight these keywords?"

@TransferHee @seongyeon01

**Comment by bw-99** [comment]
[core_concept_in_ml.csv](https://github.com/user-attachments/files/18292857/core_concept_in_ml.csv)

**Comment by bw-99** [comment]
수정된 실행 방법 (자세한건 테크스펙 참고)

0. 프로젝트 폴더에 example 폴더 생성
1. onepiece_rag 폴더 안에 input을 example 폴더로 복사 (`onepiece_rag/input` -> `example/input`)
2. `python3 -m graphrag init --root ./example`
3. `python3 util/debugging_data.py --root example --num_example 1`
4. `python3 -m graphrag index --root ./example`
5. example/output/create_final_viztree 파일 pandas로 열기

@TransferHee @seongyeon01
